pkg/api/handlers/projectmcp.go

```diff
@@ -171,11 +193,21 @@ func (p *ProjectMCPHandler) CreateServer(req api.Context) error {
 		}
 	}
 
+	var cred map[string]string
+	if mcpServer.Spec.SharedWithinMCPCatalogName == "" {
+		gptscriptCred, err := req.GPTClient.RevealCredential(req.Context(), []string{fmt.Sprintf("%s-%s", mcpServer.Spec.UserID, mcpServer.Name)}, mcpServer.Name)
+		if err != nil && !errors.As(err, &gptscript.ErrNotFound{}) {
+			return fmt.Errorf("failed to find credential: %w", err)
+		}
+
+		cred = gptscriptCred.Env
```

Comment:
are you handling err not found? if it is not found, would gptscriptCred be nil and causing a panic?

Reply:
We don't need to handle the err not found. `gptscriptCred` is not a pointer, so it will just have the zero values for everything if no cred was found, and `gptscriptCred.Env` will be a nil map that we assign to `cred`, so `cred` just remains nil.